### PR TITLE
Merge DAPS Spec with master

### PR DESCRIPTION
**Pull request by ssteinbuss**
Can we now merge this?

### CODE REVIEW

**Comment by sebbader** [comment]
Hello everyone! Great work and a lot of effort, very much appreciated. Of course I have something to complain:
* JWT is mentioned in the text but can we include a general statement like: The DAT **is** a JWT with specific keys in the Payload Part, that is also defined by the Information Model.
* A few TODOs survived:
- grafik
grafik
- and a third one for which I just lost the screenshot

File: core/DAPS/README.md [hunk 1]
```diff
@@ -1,16 +1,212 @@
 # Dynamic Attribute Provisioning Service (DAPS)
 
+The infrastructure component "Digital Attribute Provisioning Service" (DAPS)
```
Comment (tmberthold) [inline core/DAPS/README.md:3]: "Digital Attribute Provisioning Service" -> should be "Dynamic Attribute Provisioning Service"

File: core/DAPS/README.md [hunk 2]
```diff
@@ -1,16 +1,212 @@
...
+
 See also:
 - [Glossary "Dynamic Attribute Provisioning Service"](../../glossary/README.md#dynamic-attribute-provisioning-service)
 - Shortcut: `DAPS`
 
+---
+
+## Unique Identifier
+
+Each (TODO: COMPONENT/entry) connector in the IDS needs a valid, outlasting and unique
```
Comment (tmberthold) [inline core/DAPS/README.md:33]: There is a TODO in the text

File: core/DAPS/README.md [hunk 3]
```diff
@@ -1,16 +1,212 @@
 # Dynamic Attribute Provisioning Service (DAPS)
 
+The infrastructure component "Digital Attribute Provisioning Service" (DAPS)
+ in a given IDS ecosystem enables enrichment of identities of organisations
+ and connectors with additional attributes.
+ 
+DAPS can be understood (like the [Certification Authority](../CA/README.md), too) as building block of the construct of an
```
Comment (vdakker) [inline core/DAPS/README.md:7]: Should be "as a building block", sentence is a bit hard to parse.
Maybe "Like the Certification Authority, the DAPS can be understood as a building block of an IDS Identity Provider." instead?

File: core/DAPS/README.md [hunk 4]
```diff
@@ -1,16 +1,212 @@
...
+ of embedding them into [X.509 certificates](https://en.wikipedia.org/wiki/X.509)
+ are:
+
+- Attribute revocation does not enforce certificate revocation and re-issuance.
+ This is also true if new attributes are added. 
+- By defining scopes or attribute sets, only the needed attributes can be
+ included in the DAT. This limits information leakage, so only needed
+ attributes are communicated.
+- Baseline certificates can be issued to enable connector deployment
+ in uncomplicated manner.
```
Comment (vdakker) [inline core/DAPS/README.md:21]: should be "in an uncomplicated manner"

File: core/DAPS/README.md [hunk 5]
```diff
@@ -1,16 +1,212 @@
...
+
+The architecture aims to be open to multiple [Certification Authorities](../CA/README.md)
+ (CAs) issuing certificates. This means, a truly unique identifier needs to consist of
+  the issuer of the certificate and the subject identifier. For an easy machine readable
+  identifier, two ´X.509v3´ extensions will be used:
+
+### Subject Key Identifier (SKI)
+### Authority Key Identifier (AKI)
+
+The concatenation of ´SKI´ and ´AKI´ provides a unique identifier valid - even if multiple
```
Comment (vdakker) [inline core/DAPS/README.md:44]: What does this mean?

File: core/DAPS/README.md [hunk 6]
```diff
@@ -1,16 +1,212 @@
...
+
+## Request call to get a token
+
+`client_assertion` id the base64 encoded request token, shown under
+ ["Request token that is handed in at DAPS side"](#request-token-that-is-handed-in-at-daps-side).
+
+
+|**FormBody Attribute**|**content**
+|:---|:---|
+|**`grant_type`**  | OAuth based grant type. See [TODO : client_credentials grant](TODO). |
```
Comment (tmberthold) [inline core/DAPS/README.md:104]: TODO?!

File: core/DAPS/README.md [hunk 7]
```diff
@@ -1,16 +1,212 @@
...
+|**`typ`**             | yes    | 1    | The token type. Must be "JWT". |
+|**`kid`**             | yes    | 1    | Key id used to sign that token. Must match the jwks.json entry found at daps-url/.well-known/jwks.json |
+|**`alg`**             | yes    | 1    | Algorithm used to sign the token.   |
+
+The DAT has these payload fields:
+
+|**Field name**|**mandantory**|**cardinality**|**content**
+|:---|:---|:---|:---|
+|**`@context`**             | yes    | 1    | The JSON-LD context containing the IDS classes, properties and instances. Must be "https://w3id.org/idsa/contexts/context.jsonld". |
+|**`@type`**                | yes    | 1    | In the context of the IDS, the request payload is an RDF instance and therefore must state that it has "@type" : "ids:DatRequestToken". TODO: @SBA: We need a type def for this... |
```
Comment (tmberthold) [inline core/DAPS/README.md:150]: TODO in text

File: core/DAPS/requests/README.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# DAPS requests
+
+|**request**|**example**|**summary**|
+|:---|:---|:---|
+|**[POST root](./root_POST.md)** |                                       | get Dynamic Attribute Token (DAT) |
+|                                | [1 request](./root_POST.md#1-request) | get Dynamic Attribute Token (DAT) |
+||||
+
+---
```
Comment (tmberthold) [inline core/DAPS/requests/README.md:9]: This is not finished, is it?

File: core/InformationModel/Message/README.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+|**[ids:securityToken](#idssecuritytoken)**|x|1|                        
+|||
+
+---
+
+### ids:correlationMessage
+
+Correlated message. Usually needed, if a messages responds to a previous
+
+> TODO: MessageProcessedNotification >>> MessageProcessedNotification?
```
Comment (tmberthold) [inline core/InformationModel/Message/README.md:23]: TODO

File: core/InformationModel/Message/README.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+ Optional properties are supported but not required.
+
+
+### Table of Message Types
+
+| **Message Class** | Sub Class | Message Name |
+|:---|:---|:---|
+| **[Request Messages](#request-messages)**            |                                 | [ids:RequestMessage](#idsrequestmessage)
+| **[Response Messages](#response-messages)**            |                               | [ids:ResponseMessage](#idsresponsemessage)
+| **[Notification Messages](#notification-messages)**  | [ids:ConnectorNotificationMessage](#idsconnectornotificationmessage) | [ids:ConnectorAvailableMessage](#idsconnectoravailablemessage)
```
Comment (tmberthold) [inline core/InformationModel/Message/README.md:63]: Link also provides ids:MessageProcessedNotificationMessage, missing here?

File: core/InformationModel/Message/README.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+---
+
+### Request Messages
+
+##### ids:RequestMessage
+
+##### ids:DescriptionRequestMessage
+
+Message requesting metadata. If no URI is supplied via the ids:requestedElement field,
+ this messages is treated like a self-description request and the recipient should
```
Comment (vdakker) [inline core/InformationModel/Message/README.md:75]: "messages" -> "message"

File: core/InformationModel/Message/README.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+All mandatory property declarations of the core ids:Message above still hold.
+ Optional properties are supported but not required.
+
+
+### Table of Message Types
+
+| **Message Class** | Sub Class | Message Name |
+|:---|:---|:---|
+| **[Request Messages](#request-messages)**            |                                 | [ids:RequestMessage](#idsrequestmessage)
```
Comment (tmberthold) [inline core/InformationModel/Message/README.md:61]: List also provides ids:DescriptionRequestMessage, missing here?

File: core/InformationModel/Message/requests/DescriptionRequestMessage_POST.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
+# DescriptionRequestMessage POST
```
Comment (tmberthold) [inline core/InformationModel/Message/requests/DescriptionRequestMessage_POST.md:1]: not finished (does it belong to the pull-request-topic?)

File: core/InformationModel/README.md [hunk 14]
```diff
@@ -1,6 +1,12 @@
 # IDS Information Model
 
-- [Industrial Data Space Information Model](https://github.com/International-Data-Spaces-Association/InformationModel/blob/develop/README.md)
+## ids:Message
```
Comment (tmberthold) [inline core/InformationModel/README.md:3]: ?

File: core/InformationModel/Message/requests/README.md [hunk 15]
```diff
@@ -0,0 +1,9 @@
+# Message requests
+
```
Comment (tmberthold) [inline core/InformationModel/Message/requests/README.md:2]: ?

File: core/MetaDataBroker/README.md [hunk 16]
```diff
@@ -3,3 +3,15 @@
 Shortcut: `IDS-MDB`
```
Comment (tmberthold) [inline core/MetaDataBroker/README.md:3]: from now on are classes, which dont belong to the DAPS topic (which should be the pull-request)

File: core/DAPS/README.md [hunk 17]
```diff
@@ -1,16 +1,212 @@
...
+ identifier, never be re-used for any other resource inside the IDS ecosystem.
+
+The architecture aims to be open to multiple [Certification Authorities](../CA/README.md)
+ (CAs) issuing certificates. This means, a truly unique identifier needs to consist of
+  the issuer of the certificate and the subject identifier. For an easy machine readable
+  identifier, two ´X.509v3´ extensions will be used:
+
+### Subject Key Identifier (SKI)
+### Authority Key Identifier (AKI)
+
```
Comment (tmberthold) [inline core/DAPS/README.md:43]: The way the headings have been placed it looks like a description should be added below them, but isn't yet.

File: core/DAPS/requests/root_POST.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+
+
+### 1 response
+
+The response to given request is a
+ [Dynamic Attribute Token (DAT)](../README.md#dynamic-attribute-token-dat), here shown decoded:
+
+```
+{
+    "typ": "JWT",
```
Comment (vdakker) [inline core/DAPS/requests/root_POST.md:51]: "kid" header is missing?

File: core/DAPS/README.md [hunk 19]
```diff
@@ -1,16 +1,212 @@
...
+|**`@context`**  | yes   | 1 | The JSON-LD context containing the IDS classes, properties and instances. Must be "https://w3id.org/idsa/contexts/context.jsonld". |
+|**`@type`**     | yes   | 1 | In the context of the IDS, the request payload is an RDF instance and therefore must state that it has "@type" : "ids:DatRequestToken". |
+|**`sub`**       | yes   | 1 | Subject the requesting connector the token is created for. This is the connector requesting the DAT. The "sub" value must be the  combined entry of the SKI and AKI of the IDS X509 as presented in Sec. 4.2.1.  In this context, this is identical to "iss". |
+|**`exp`**       | yes   | 1 | Expiration date of the token. Can be chosen freely but should be limited to a short period of time (e.g., one minute). |
+|**`iat`**       | yes   | 1 | Timestamp the token has been issued. |
+|**`nbf`**       | yes   | 1 | "Valid not before": For practical reasons this should be identical to iat. If the system time is not in synch with the DAPS, the request token will be rejected (e.g., nbf is in the future). |
+|**`aud`**       | yes   | 1 | The audience of the token. This can limit the validity for certain connectors. |
```
Comment (schanzen) [inline core/DAPS/README.md:91]: The audience of the request token must be the target DAPS. For example "https://daps.aisec.fraunhofer.de". It is the same value as the "iss" for the resulting DAT. The "aud" value is basically an indication of who the token is for. The request token is not for connectors, it is for the DAPS.

File: core/DAPS/README.md (hunk 6 above)
Comment (schanzen) [reply core/DAPS/README.md:104]: This should probably not be a TODO. The grant type is (usually) "client_credentials" (https://tools.ietf.org/html/rfc6749#section-4.4)

File: core/DAPS/requests/root_POST.md (hunk 18 above)
Comment (schanzen) [reply core/DAPS/requests/root_POST.md:51]: Note that the "kid" value (as all other JWT-specific claims) are non normative. But most implementations will issue JWTs with "kid" fields.

File: core/DAPS/README.md (hunk 1 above)
Comment (gbrost) [reply core/DAPS/README.md:3]: Absolutely right. Wherever this digital came from...

File: core/DAPS/README.md (hunk 3 above)
Comment (gbrost) [reply core/DAPS/README.md:7]: Sounds better.

**Review by gbrost — CHANGES_REQUESTED** [review]

File: core/DAPS/requests/README.md (hunk 8 above)
Comment (jlangkau) [reply core/DAPS/requests/README.md:9]: This is ONLY a suggestion from my side to get an idea, how to show an request (in a generic manner...). You can see it under Broker-stuff, also...

File: core/DAPS/README.md (hunk 2 above)
Comment (jlangkau) [reply core/DAPS/README.md:33]: All those "TODOs" are TODOs indeed. At the time I started those entries (keep in mind: DRAFT!!!) I tried to learn and show how it possible could look like - nothing more. The DAPS/DAT was the first one I'd started with. So those TODOs were very natural, to say so from todays perspective ;-) In the future we we will fight those TODOs in IDS-G-pre!!!

File: core/DAPS/README.md (hunk 17 above)
Comment (jlangkau) [reply core/DAPS/README.md:43]: Yes, maybe something has to be added here - keep in mind: DRAFT-branch!!!

File: core/DAPS/README.md (hunk 5 above)
Comment (jlangkau) [reply core/DAPS/README.md:44]: Hm, bad explanation?!? Uniquness of identifier, even from different domain/sources aka CAs...

File: core/DAPS/README.md (hunk 6 above)
Comment (jlangkau) [reply core/DAPS/README.md:104]: TODO means TODO. DRAFT-branch!

File: core/DAPS/README.md (hunk 7 above)
Comment (jlangkau) [reply core/DAPS/README.md:150]: TODO means TODO. DRAFT-branch!

File: core/InformationModel/Message/README.md (hunk 12 above)
Comment (jlangkau) [reply core/InformationModel/Message/README.md:61]: DRAFT-branch!!!! Those days I'd tried to get in touch with the ids:Messages - but noc one from IDS-IM was able to help me to lift it to a better level... maybe nowadays?!?

File: core/DAPS/README.md (hunk 7 above)
Comment (tmberthold) [reply core/DAPS/README.md:150]: Well, I should review if the DRAFT-Branch-PullRequest can in the meantime be merged into the master, whereby that would also be on the master ;)

File: core/InformationModel/Message/README.md (hunk 10 above)
Comment (jlangkau) [reply core/InformationModel/Message/README.md:63]: DRAFT-branch. this was a sketch only!

File: core/InformationModel/Message/requests/DescriptionRequestMessage_POST.md (hunk 13 above)
Comment (jlangkau) [reply core/InformationModel/Message/requests/DescriptionRequestMessage_POST.md:1]: DRAFT-branch. this was a sketch only!

File: core/InformationModel/Message/requests/README.md (hunk 15 above)
Comment (jlangkau) [reply core/InformationModel/Message/requests/README.md:2]: DRAFT-branch. this was a sketch only!

File: core/DAPS/README.md (hunk 5 above)
Comment (vdakker) [reply core/DAPS/README.md:44]: 'The concatenation of ´SKI´ and ´AKI´ provides a unique identifier valid - ...' sounds like there is something missing.

so it is meant like: 

'The concatenation of ´SKI´ and ´AKI´ provides an identifier which is unique - even if multiple CAs are able to issue valid certificates' ?
